x-pack/plugins/stack_alerts/server/alert_types/es_query/alert_type_params.ts

```diff
@@ -21,13 +22,19 @@ export interface EsQueryAlertState extends RuleTypeState {
   latestTimestamp: string | undefined;
 }
 
+export type EsQueryAlertParamsExtractedParams = Omit<EsQueryAlertParams, 'searchConfiguration'> & {
+  searchConfiguration: SerializedSearchSourceFields & {
+    indexRefName: string;
+  };
+};
+
 const EsQueryAlertParamsSchemaProperties = {
   size: schema.number({ min: 0, max: ES_QUERY_MAX_HITS_PER_EXECUTION }),
   timeWindowSize: schema.number({ min: 1 }),
   timeWindowUnit: schema.string({ validate: validateTimeWindowUnits }),
   threshold: schema.arrayOf(schema.number(), { minSize: 1, maxSize: 2 }),
   thresholdComparator: getComparatorSchemaType(validateComparator),
-  searchType: schema.nullable(schema.literal('searchSource')),
+  searchType: schema.oneOf([schema.literal('searchSource'), schema.literal('esQuery')]),
```

Comment:
One last add-on, let's set a default value here. Then it won't break for users using REST to create their rule.
```suggestion
  searchType: schema.oneOf([schema.literal('searchSource'), schema.literal('esQuery')], {
    defaultValue: 'esQuery',
  }),
```